tools/meteor-services/deploy.js

```diff
@@ -321,6 +323,206 @@ function canonicalizeSite(site) {
+    // The deploy succeeded and the deployed version is now active
+    const {newVersionNum, hostname, username, region, galaxyUrl} = result.payload;
+    var successfulDeploymentMessage = `
+    ******************************************************************************
+    You have successfully upgraded your app to version ${newVersionNum}.
+    Hostname: ${hostname}, Region: ${region}, Username: ${username}
+    To see the status of your app, visit your app's page on Galaxy
+    \t\t${galaxyUrl}.
+    ******************************************************************************
+    `
+    Console.info(successfulDeploymentMessage);
+  } else {
+    // The deploy did not succeed or the polling timed out
+    if(deploymentPollResult.buildStatus === 'failed-permanently' ||
+      deploymentPollResult.deployStatus === 'failed') {
+        const {newVersionNum, hostname, username, region, galaxyUrl} = result.payload;
+        var failedDeploymentMessage = `
+        ******************************************************************************
+        Failed to deploy your app to version ${newVersionNum}. No changes were made.
+        Hostname: ${hostname}, Region: ${region}, Username: ${username}
+        To see the logs for your app, visit ${galaxyUrl}/logs.
+        ******************************************************************************
+        `
+        Console.info(successfulDeploymentMessage);
+      } else {
+        // The status was non-terminal, so we most likely timed out
+        Console.info(result.payload.message);
+        return 1;
+      }
+  }
+}
+
+// Creates a polling configuration with defaults if fields left unset
+// Default initialWaitTime is 10 seconds – this is the time to wait before checking at all
+// Default startInterval is 700 milliseconds – this is the first wait interval between polls
+// Default backoffFactor is 1.3
+// Default timeout is 15 minutes
+function PollingConfiguration(
+  timeoutMillis,
+  initialWaitTimeMillis,
+  startIntervalMillis,
+  backoffFactor) {
+    const fifteenMinutesMillis = 15*60*1000;
+    this.initialWaitTimeMillis = initialWaitTimeMillis || 10*1000;
+    this.startIntervalMillis = startIntervalMillis || 700;
+    this.backoffFactor = backoffFactor || 1.3;
+    this.timeoutMillis = timeoutMillis || fifteenMinutesMillis;
+  }
+
+// Build statuses that indicate we should stop polling for build
+const terminalBuildStatuses = ['failed-permanently', 'success']
+// Deploy statuses that indicate we should stop polling for deploy
+const terminalDeployStatuses = ['success', 'failed', 'aborted']
+
+// Declare tracking variables for the duration of the poll
+// `start` tracks the time when we started polling so we can know when to time out
+// `pollingRounds` tracks the number of rounds we have polled, resetting when build/deploy succeed
+// `currentStatus` tracks what the current status is of the build and the deploy for this version
+let start, pollingRounds, currentStatus;
```

Comment:
Even though we're unlikely to have parallel deploys in the tool, global mutable state is not a great idea. It seems like these could easily go on your polling configuration object (which could be renamed to PollingState or something).